toolkit/src/styles/components/list-item.scss

```diff
@@ -0,0 +1,54 @@
+/**
+* Copyright (c) HashiCorp, Inc.
+*/
+
+@import '@hashicorp/design-system-components';
+
+.cut-list-item {
+  display: flex;
+  flex-direction: row;
+  flex-grow: 0;
+  justify-content: space-between;
+  align-items: center;
+  padding: 12px;
+  border-radius: 6px;
+  gap: 8px;
+}
+
+button.cut-list-item.active:focus,
+a.cut-list-item.active:focus,
+button.cut-list-item.active:focus-visible,
+a.cut-list-item.active:focus-visible
+{
+  text-decoration: none;
+  outline: 3px solid var(--token-color-focus-action-external);
+  outline-offset: 1px;
+}
+
+a.cut-list-item {
+  text-decoration: none;
+  color: inherit;
+}
+
+button.cut-list-item {
+  background-color: unset;
+  width: 100%;
+  border: inherit;
+  font-family: inherit;
+  font-size: inherit;
+  color: inherit;
+}
+
+.cut-list-item.active:hover {
+  box-shadow: var(--token-surface-high-box-shadow);
+}
+
+.cut-list-item__content {
+  display: flex;
+  justify-content: start;
+  flex: 1;
+}
+
+.cut-list-item.active .cut-list-item__content {
+  cursor: pointer;
```

Comment:
can you explain the rationale behind this styling? it doesn't sound right to me 🤔